src/__tests__/to-be-disabled.js

```diff
@@ -10,92 +9,77 @@ import {
 } from 'react-native';
 import { render } from '@testing-library/react-native';
 
-test('.toBeDisabled', () => {
-  const { queryByTestId, queryByText, queryByTitle, queryByDisplayValue } = render(
-    <View disabled testID="view">
-      <Button disabled testID="button" title="button" />
-      <TextInput editable={false} testID="textInput" value="textInput" />
-      <TouchableHighlight disabled testID="highlight">
-        <Text>highlight</Text>
-      </TouchableHighlight>
-      <TouchableOpacity disabled testID="opacity">
-        <Text>opacity</Text>
-      </TouchableOpacity>
-      <TouchableWithoutFeedback disabled testID="without">
-        <Text>without</Text>
-      </TouchableWithoutFeedback>
-    </View>,
-  );
-
-  expect(queryByTestId('view')).toBeDisabled();
-  expect(() => expect(queryByTestId('view')).not.toBeDisabled()).toThrowError();
-
-  expect(queryByTitle('button')).toBeDisabled();
-  expect(() => expect(queryByTitle('button')).not.toBeDisabled()).toThrowError();
-
-  expect(queryByTestId('textInput')).toBeDisabled();
-  expect(queryByDisplayValue('textInput')).toBeDisabled();
-  expect(() => expect(queryByTestId('textInput')).not.toBeDisabled()).toThrowError();
-  expect(() => expect(queryByDisplayValue('textInput')).not.toBeDisabled()).toThrowError();
-
-  expect(queryByTestId('highlight')).toBeDisabled();
-  expect(queryByText('highlight')).toBeDisabled();
-  expect(() => expect(queryByTestId('highlight')).not.toBeDisabled()).toThrowError();
-  expect(() => expect(queryByText('highlight')).not.toBeDisabled()).toThrowError();
-
-  expect(queryByTestId('opacity')).toBeDisabled();
-  expect(queryByText('opacity')).toBeDisabled();
-  expect(() => expect(queryByTestId('opacity')).not.toBeDisabled()).toThrowError();
-  expect(() => expect(queryByText('opacity')).not.toBeDisabled()).toThrowError();
-
-  expect(queryByTestId('without')).toBeDisabled();
-  expect(queryByText('without')).toBeDisabled();
-  expect(() => expect(queryByTestId('without')).not.toBeDisabled()).toThrowError();
-  expect(() => expect(queryByText('without')).not.toBeDisabled()).toThrowError();
+const ALLOWED_COMPONENTS = new Map([
+  ['View', View],
+  ['Button', Button],
+  ['TextInput', TextInput],
+  ['TouchableHighlight', TouchableHighlight],
+  ['TouchableOpacity', TouchableOpacity],
+  ['TouchableWithoutFeedback', TouchableWithoutFeedback],
+]);
```

Comment:
Why not just using an object? 🤔

```suggestion
const ALLOWED_COMPONENTS = {
  View,
  Button,
  TextInput,
  TouchableHighlight,
  TouchableOpacity,
  TouchableWithoutFeedback,
};
```

Reply:
Good catch. thank you.